### ISSUE TITLE
plotting comparison with a control

### ISSUE THREAD

**Issue by rcprati** [b0rxa/scmamp#1]
Hi,

Thanks for the package.

I've tried without success to use plotRanking for plotting the results of a comparison with a control. Is there any way to do that?

Regards

**Comment by b0rxa** [comment]
Hi,

The plotRanking is a figure "inspired" in the critical differences plot's presented in Demsâr's papers. Those plots were designed to compare all vs. all the algorithms, so the plotRanking are also designed for that. 

If you want to do something similar, my suggestion is to create a square matrix similar to those obtained when comparing all vs all, fill it with 0 (meaning that the differences are significant) and the put the p-values of the comparison between all the algorithms and the control in BOTH the row and the column corresponding to the control. It is an artificial p-value matrix where the comparisons with the control are the true ones and any other comparison is regarded as significantly different. 

Just one warning. Such a plot can be misleading if anybody interprets it as an all vs. all comparison, because you are indicating that there are differences between every pair of algorithms that do not include the control but, actually, you have not tested that. Therefore, my suggestion is that if you want to use it, clearly state that only the comparisons with the control have been included.

Regards,

**Comment by rcprati** [comment]
Thanks for the answer. 

In fact, the Demsar's paper shows two kind of plots, and I was thinking in a plot similar to figure 1(b) in the paper, where the plots are draw with respect to a control: 

![screen shot 2016-12-02 at 11 02 41 am](https://cloud.githubusercontent.com/assets/3610963/20830010/fb33dcde-b87e-11e6-9083-d55a5bc6146b.png)

I'll try out our suggestion,  and when I have some spare time try to write a function for doing that.

Regards,


Ronaldo

**Comment by b0rxa** [comment]
Yes, you are right. We did not implemented that option, but I have already updated the github plotting function to include it. As an example, now this should work:

res <- postHocTest(data.gh.2008, control=1, use.rank=TRUE)
plotRanking(res$corrected.pval, res$summary)

Thanks for pointing this issue out. Let me know if you have any problem.

Regards,

**Comment by rcprati** [comment]
Thanks for that. BTW, it was not an issue, but a feature :)

However, I'm getting the following error trying to install from github:

Error in eapply(ns_env(pkg), force, all.names = TRUE) : 
  lazy-load database '/Library/Frameworks/R.framework/Versions/3.3/Resources/library/scmamp/R/sysdata.rdb' is corrupt

**Comment by rcprati** [comment]
OOPS, it seems to be a problem with my local environment. I restarted R, and I could successfully install. Thanks again for the new feature and the package.

Regards,

Ronaldo
